test/integration/sanity/clear-vars-sandbox.test.js

```diff
@@ -0,0 +1,112 @@
+                        script: {
+                            type: 'text/javascript',
+                            exec: 'postman.setEnvironmentVariable(\'e\', \'2\');\npostman.setGlobalVariable(\'g\',\'3\');'
+                        }
+                    }],
+                    request: {
+                        url: 'https://postman-echo.com/get?a={{e}}&b={{g}}',
+                        method: 'GET',
+                        header: [],
+                        body: {
+                            mode: 'formdata',
+                            formdata: []
+                        },
+                        description: 'rd'
+                    },
+                    response: []
+                }, {
+                    name: 'After clearing',
+                    event: [{
+                        listen: 'test',
+                        script: {
+                            type: 'text/javascript',
+                            exec: 'var data = JSON.parse(responseBody);\ntests[\'Variable substitution from env not working after clearing\'] = (data.args.a===\'{{e}}\');\ntests[\'Variable substitution from global not working after clearing\'] = (data.args.b===\'{{g}}\');\npostman.setEnvironmentVariable(\'e\', \'2\');\npostman.setGlobalVariable(\'g\',\'3\');\ntests[\'Recreated vars\']=environment.e===\'2\' && globals.g===\'3\'\npostman.clearEnvironmentVariable(\'e\');\npostman.clearGlobalVariable(\'g\');'
+                        }
+                    }],
+                    request: {
+                        url: 'https://postman-echo.com/get?a={{e}}&b={{g}}',
+                        method: 'GET',
+                        header: [],
+                        body: {
+                            mode: 'formdata',
+                            formdata: []
+                        },
+                        description: 'rd'
+                    },
+                    response: []
+                }, {
+                    name: 'After clearing',
+                    event: [{
+                        listen: 'test',
+                        script: {
+                            type: 'text/javascript',
+                            exec: 'console.log(environment);console.log(globals);var data = JSON.parse(responseBody);\ntests[\'Variable substitution from env not working after clearing invididually\'] = (data.args.a===\'{{e}}\');\ntests[\'Variable substitution from global not working after clearing invididually\'] = (data.args.b===\'{{g}}\');'
+                        }
+                    }],
+                    request: {
+                        url: 'https://postman-echo.com/get?a={{e}}&b={{g}}',
+                        method: 'GET',
+                        header: [],
+                        body: {
+                            mode: 'formdata',
+                            formdata: []
+                        },
+                        description: 'rd'
+                    },
+                    response: []
```

Comment:
All such unwanted arrays can be removed, and the collection need not seed it's environment variables prior to a test. A separate environment file can be passed for better checks, with the resultant `environment` global being checked post run.